packages/headless/src/database/mongodb/product/index.ts

```diff
@@ -55,4 +58,46 @@ export class ProductDatabase implements IProductDatabase {
   async getProductsList(skip: number, limit: number, query?: Record<string, any>, sortCondition?: string): Promise<Product[] | []> {
     return await ProductModel.find(query, '-_id').sort(sortCondition).skip(skip).limit(limit).lean();
   }
-}
\ No newline at end of file
+
+  async getTags(query: Record<string, any>): Promise<Tag[]> {
+    return await TagsModel.find(query).select('-_id').lean();
+  }
+
+  async getTopSellingProducts( skip: number, limit: number): Promise<Product[] | []> {
```

Comment:
Seems it will increase load in database for this aggregation. lots of data will be populated.
Rather it would be better to keep track of last let's say 500 orders product data in a separate collection. don't insert the whole order as a document, rather split the order items as separate entity and insert them, so that you don't have to unwind again and again.
How will you ensure that only last 500 data will remain in that collection?
Refer to this: https://www.mongodb.com/docs/v6.0/core/capped-collections/